### ISSUE TITLE
[BUG] Multiple Authentication Types applied together are not handled correctly

### ISSUE THREAD

**Issue by zdne** [superfaceai/cli#363]
When the Open API specification defines multiple authentication types and their are applied simultaneously the provider definition defines them as two separate authentication types.

Refer to [Using Multiple Authentication Types](https://swagger.io/docs/specification/authentication/). 

In the situation where security is defined as two types applied at the same type

```yaml
security:    # A AND B are applied together
  - A
    B
```

for example:

```json
  "security": [
    {
      "basicAuth": [],
      "instanceName": []
    }
  ],
```

with security schemes defined as follows: 

```json
  "components": {
    "securitySchemes": {
      "basicAuth": {
        "type": "http",
        "scheme": "basic"
      },
      "instanceName": {
        "type": "apiKey",
        "description": "instance name",
        "name": "X-Instance-Name",
        "in": "header"
      }
    },
```

the resulting security of APIs should include **both** basic authentication and the custom "api key" header `X-Instance-Name`. Without either one of those the call is considered unauthenticated

## Expected Behavior
The two authentication types are used together.

## Possible Solution
We will most likely have to change the provider definition format to accommodate multiple auth types used simultaneously.

## Steps to Reproduce
1. Visit https://app.raynet.cz/api/doc/
2. Download their OAS
3. Run `superface prepare <raynet OAS>

## Your Environment
- @superfaceai/cli/4.0.1 darwin-arm64 node-v18.13.0

**Comment by Jakub-Vacek** [comment]
I think this is more limitation that a bug. There is definitely some work in Engine that needs to be done to support this. 
@freaz Can SDK use more that one security id?

**Comment by freaz** [comment]
You are right. OneSDK never supported applying multiple security configurations to one request, so it needs to be added.
